### ISSUE TITLE
Fix S1144 FP: Event with a concrete sender

### ISSUE THREAD

**Issue by IeuanWalker** [SonarSource/sonar-dotnet#6990]
### Description

If I create an event with a concreate sender it gives the warning S1144 _(remove unused code)_
i.e. this is gives a warning  -
```csharp
void CustomSwitch_SwitchPanUpdate(CustomSwitch customSwitch, SwitchPanUpdatedEventArgs e)
```
but this is fine -
```csharp
void CustomSwitch_SwitchPanUpdate(object customSwitch, SwitchPanUpdatedEventArgs e)
```

### Repro steps

In the `CustomSwitch` class i have this -
```csharp
public delegate void CustomSwitchPanUpdatedEventHandler(CustomSwitch customSwitch, SwitchPanUpdatedEventArgs e);
public event CustomSwitchPanUpdatedEventHandler? SwitchPanUpdate;
```
And I trigger it like this -
```csharp
SwitchPanUpdate?.Invoke(this, ev);
```

Then in a consuming class I have this -
```csharp
void CustomSwitch_SwitchPanUpdate(CustomSwitch customSwitch, SwitchPanUpdatedEventArgs e)
{
	// Logic
}
```

### Expected behaviour

No warning, first parament should be ignored and can be anything not just `object`

### Actual behaviour

Warning

### Known workarounds

Setting the first parameter as `object`

**Comment by martin-strecker-sonarsource** [comment]
Hello @IeuanWalker,

Thank you for reporting this. Can you please tell if the `SwitchPanUpdatedEventArgs` class derives from [`EventArgs`](https://learn.microsoft.com/dotnet/api/system.eventargs)? If so, we could relax the exception described in the S1144 rule as `Methods with event handler signature void Foo(object, EventArgs) that are declared in partial class` to also allow concrete classes for the `sender` parameter.

**Comment by IeuanWalker** [comment]
@martin-strecker-sonarsource hi, ye it is - https://github.com/IeuanWalker/Maui.Switch/blob/master/Scr/Switch/Events/SwitchPanUpdatedEventArgs.cs

**Comment by martin-strecker-sonarsource** [comment]
Hi @IeuanWalker,

We discussed the issue and decided to mark this as a false positive. The S1144 rule exception reads like this now.

> Methods with event handler signature `void Foo(object, EventArgs)` that are declared in partial class 

The method signature requirement should be relaxed to

> `void` returning method with two parameters, and the second parameter should derive from `EventArgs`

We move this issue to our backlog, and you can track our work here.
